Bash-Release: 3.2
Patch-ID: bash32-008

Bug-Reported-by:    Linda Walsh <bash@tlinx.org>
Bug-Reference-ID:   <456041FD.8000605@tlinx.org>
Bug-Reference-URL: http://lists.gnu.org/archive/html/bug-bash/2006-11/msg00040.html

Bug-Description:

When checking pathnames from the command hash table (e.g., when the checkhash'
shell option is enabled), a bug causes bash to delete and re-lookup each command.

--- bash-3.2/findcmd.c	2005-08-17 22:49:54.000000000 +0200
+++ bash-3.2/findcmd.c.new	2007-04-01 03:15:01.000000000 +0200
@@ -308,7 +308,7 @@
   if (hashed_file && (posixly_correct || check_hashed_filenames))
     {
       st = file_status (hashed_file);
-      if ((st ^ (FS_EXISTS | FS_EXECABLE)) != 0)
+      if ((st & (FS_EXISTS|FS_EXECABLE)) != (FS_EXISTS|FS_EXECABLE))
 	{
 	  phash_remove (pathname);
 	  free (hashed_file);
